### PR TITLE
interpolation coefs components and tests

### PR DESCRIPTION
**Pull request by ayashunsky**
These are simple components for computing coefficients of linear and quadratic interpolations

### CODE REVIEW

**Review by Iluvmagick — CHANGES_REQUESTED** [review]
The components are good!
But please add the test to `run_tests.yml` so that they run in our CI, and rebase on master.

File: include/nil/blueprint/components/algebra/fields/plonk/quadratic_interpolation.hpp [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+#include <nil/blueprint/component.hpp>
+#include <nil/blueprint/manifest.hpp>
+
+namespace nil {
+    namespace blueprint {
+        namespace components {
+            namespace detail {
+                // compute the determinant of a 3x3 matrix
+                template<typename T>
+                T det3(std::array<T,9> a) {
```
[the remark this answers is not in the document]
Comment (ayashunsky) [reply include/nil/blueprint/components/algebra/fields/plonk/quadratic_interpolation.hpp:43]: initializer_list doesn't support the [] operator unfortunately. For now, leave it as it is.

File: test/algebra/fields/plonk/interpolation.cpp [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+    static boost::random::mt19937 seed_seq;
+    static nil::crypto3::random::algebraic_engine<BlueprintFieldType> generate_random(seed_seq);
+
+    for (std::size_t i = 0; i < RandomTestsAmount; i++){
+        test_linear_inter_coefs<BlueprintFieldType>(
+            {generate_random(), generate_random(), generate_random(), generate_random()});
+        test_quadratic_inter_coefs<BlueprintFieldType>(
+            {generate_random(), generate_random(), generate_random(), generate_random(), generate_random(), generate_random()});
+    }
+    // one explicitely failing test
```
[the remark this answers is not in the document]
Comment (ayashunsky) [reply test/algebra/fields/plonk/interpolation.cpp:186]: Fixed

**Comment by Iluvmagick** [comment]
Either something went wrong during the rebase here, or github is bugged.
Only your four commits should be visible in the pull request.
Somehow the changes are duplicating?
Unsure what is going on.

**Review by Iluvmagick — APPROVED** [review]

**Pull request merged**